Fixes datetimepicker update by ajax call before completly initialized

mtvweb opened the pull request:
Using a b:datetimepicker with ajax update was sending an invalid (ISO)
format date to the server before initialization of the datetimepicker
itself. Moving event initialization to the jquery ready handler fixes
this behaviour.

Fix #586

stephanrauh commented:
I've tried to integrate your bug fix. At first glance, it looks great, but there seem to be at least two issues:

* I had to simplify your reproducer because I don't have your template file. After that, jQuery wasn't available, resulting in errors like "Uncaught ReferenceError: $ is not defined".
* I wonder if your `$(function() { ... }` (which is equivalent to `$( document ).ready(function() { ... });` works if the datePicker is created by an AJAX request. The difference being that `ready()` probably is never called. But I didn't test it yet, so it's just guesswork.

Nonetheless: thank you very much for providing the pull request and the great reproducer!

Best regards,
Stephan

mtvweb commented:
Sorry, I had only stripped down a part of my project to the relevant code, but forget about the layout.  I've changed it now to a standalone one. Also I added some more to test the update case. Which is working now well for me.

- Regarding the ReferenceError. Because the datetimepicker is using the same $(function() ....) some lines before for initialization in DateTimePickerRenderer, so it should be there too. Maybe something was missing and the updated test files help.

- The initialization should work, even with ajax inserted in the DOM. My experience (by JSF and non JSF projects) is that JQuery schedules them quite well, even if the document was already initialized. The advantage is that they are executed in the same sequence as they are added to the ready handler. You are using $() initialization for DateTimePicker initialization which was working well too for ajax requested ones.

Regards
Jens 

[DTPBean.java.txt](https://github.com/TheCoder4eu/BootsFaces-OSP/files/660557/DTPBean.java.txt)
[dtptest.xhtml.txt](https://github.com/TheCoder4eu/BootsFaces-OSP/files/660558/dtptest.xhtml.txt)

stephanrauh commented:
@mtvweb Sounds good! I'm sorry that your PR didn't make it into the 1.0.1 version. We thought about integrating it, but it would have really been a last minute change, so we decided to err on the safe side. There simply was no time to test it. I hope I managed to run and integrate the tests later this week.
